Cloud Application Manager/Troubleshooting/camdce-troubleshooting-tips.md

```diff
@@ -0,0 +1,289 @@
+
+**Solution 2**
+
+Check agent proxy settings (if configured to use one) at `/usr/elasticbox/elasticbox.conf` in linux or `ProgramFiles\ElasticBox\Agent\elasticbox.conf` in windows. If proxy is used for agent, make sure its working correctly and provide stable connectivity to the CAM appliance.
+
+**Solution 3**
+
+Make sure the communication from CAM Agent to CAM appliance works correctly.
+
+1. Connect to the instance by SSH or RDP.
+
+2. Run a ping command to check basic connectivity:
+
+    ```
+    ping -c3 10.0.0.1
+    ```
+
+    **Note:** Replace 10.0.0.1 with your appliance IP or hostname.
+
+If the above command does not get a reply, please:
+
+1. Take the corrective actions inside the VM, routers and firewalls to allow unrestricted communication between agent and the CAM appliance.
+
+2. In Cloud Application Manager, open the lifecycle editor of the instance and click **Reinstall**. The instance should reflect the proper status.
+
+If the connectivity seems to be working as expected and the problem persists, connect to the instance over SSH or RDP and grab a copy of the agent `/var/log/elasticbox/elasticbox-agent.log` in the case of Linux or `ProgramData\ElasticBox\elasticbox-agent.log` in the case of Windows (please note that the folder is hidden in Windows and the exact path must be entered to navigate to the folder). Once the log file is collected please attach it when [Contacting support](mailto:incident@CenturyLink.com)
+
+**Cause 2**
+
+When a box script terminates with 100 exit code, typically because of apt-get failures, the Cloud Application Manager agent goes into sleep mode waiting for the machine or service to reboot.
+
+**Solution**
+
+1. Open the instance page in CAM to review the deployment logs
+2. Identify which script is exiting with code 100.
+3. Open lifecycle editor
+4. Edit the previously identified and add the following line just after the command that exited with 100.
+
+   ```
+   [[ "$?" -eq "100" ]] && exit 1
+   ```
+
+5. Connect to Instance over SSH or RDP
+6. Restart the agent.
+
+   Linux:
+
+   ```
+   sudo /usr/elasticbox/elasticbox restart
+   ```
+
+   Windows (run in command prompt):
+
+   ```
+   net stop elasticbox
+   net start elasticbox
+   ```
+
+### Instance is Still Terminating
+
+**Cause**
+
+Several concurrent terminate requests within seconds of each other can cause a race condition and keep the instance in running state.
+
+**Solution**
+
+From CAM instance page, Force the instance to terminate, or alternatively, in the lifecycle editor of the instance, click Force Terminate.
+
+### Instance Hangs at the Install Cloud Application Manager Agent Step
+
+**Cause**
+
+Something causes the agent to hang even though it’s running on the instance.
```

Comment:
This also happens when no outbound connectivity exists (cam.ctl.io 443 is not accessible).